projects/cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,78 @@
+package cmd
+
+import (
+	"bufio"
+	_ "embed"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+)
+
+//go:embed help.txt
+var helpFile string
+
+func Execute() error {
+	fileNames, numberLine, help := ParsingFlags()
+
+	err := flagHandler(os.Stdout, fileNames, numberLine, help)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ParsingFlags() (fileNames []string, numberLine *bool, help *bool) {
+	numberLine = flag.Bool("n", false, "Add number in front of each line")
+	help = flag.Bool("h", false, "Show help")
+
+	flag.Parse()
+	fileNames = flag.Args()
+	return
+}
+
+func flagHandler(w io.Writer, fileNames []string, numberLine *bool, help *bool) error {
+	if *help {
+		fmt.Fprintln(w, string(helpFile))
+		return nil
+	}
+
+	if len(fileNames) == 0 && !*help {
+		return fmt.Errorf("did not provide any path")
+	}
+
+	for _, path := range fileNames {
+		err := OpenFile(w, path, *numberLine)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func OpenFile(w io.Writer, path string, numberLine bool) error {
```

Comment:
This function feels misleadingly named - it doesn't just _open_ a file, it also _prints_ the file?